### PR TITLE
Create draft property-fire-safety-payment Smart Answer

### PR DESCRIPTION
**Pull request by edwardkerry**
This PR adds a new Smart-answer to help users calculate how much they may need to pay for fire safety improvements to their property.

It asks users 
- building height
- if they own freehold
- if they own more than 3 properties
- (if yes) if this property was the main residence in February 2022
- year of property purchase
- price of property at purchase
- if the property is in London
- if it's in a shared ownership scheme
- (if yes) what percentage is owned

Some uprating values have been provided to estimate the current value of the property, based on the year of purchase and price at purchase.

Depending on the estimated value, percentage of property owned, and location, a figure is calculated to indicate potential costs.

This smart-answer is set to "draft", so will not be published. The copy in the views is placeholder and has not yet been through content design or review.

### CODE REVIEW

File: test/unit/calculators/property_fire_safety_payment_calculator_test.rb [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+          @calculator.stubs(:uprated_value_of_property).returns(PropertyFireSafetyPaymentCalculator::INNER_LONDON_VALUATION_LIMIT)
+          @calculator.live_in_london = "yes"
+        end
+
+        should "return fifteen thousand if not shared ownership" do
+          @calculator.shared_ownership = "no"
+          assert_equal @calculator.leaseholder_costs, 15_000
+        end
+
+        should "return pecentage owned multipled by fifteen thousand if shared ownership" do
```
Comment (KludgeKML) [inline test/unit/calculators/property_fire_safety_payment_calculator_test.rb:110]: Tiny note: here and in the other tests percentage has a typo

File: lib/smart_answer/calculators/property_fire_safety_payment_calculator.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  class PropertyFireSafetyPaymentCalculator
+    attr_accessor :year_of_purchase,
+                  :value_of_property,
+                  :live_in_london,
+                  :shared_ownership,
+                  :percentage_owned
+
+    EARLIEST_VALID_YEAR = 1945
+    OUTER_LONDON_VALUATION_LIMIT = 175_000
+    INNER_LONDON_VALUATION_LIMIT = 325_000
```
Comment (KludgeKML) [inline lib/smart_answer/calculators/property_fire_safety_payment_calculator.rb:11]: Are these actually outer london and inner london, or are they outside london and inside london?

Comment (edwardkerry) [reply lib/smart_answer/calculators/property_fire_safety_payment_calculator.rb]: Thanks Keith, you're right. I've changed them from OUTER/INNER to OUTSIDE/INSIDE

File: test/unit/calculators/property_fire_safety_payment_calculator_test.rb (hunk 1 above)
Comment (edwardkerry) [reply test/unit/calculators/property_fire_safety_payment_calculator_test.rb]: good spot 👍

File: test/unit/calculators/property_fire_safety_payment_calculator_test.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+require_relative "../../test_helper"
+
+module SmartAnswer::Calculators
+  class PropertyFireSafetyPaymentCalculatorTest < ActiveSupport::TestCase
+    setup do
+      @calculator = PropertyFireSafetyPaymentCalculator.new
+    end
+
+    context "#valid_year_of_purchase?" do
```
[the remark this answers is not in the document]
Comment (edwardkerry) [reply test/unit/calculators/property_fire_safety_payment_calculator_test.rb:9]: good call, I'm sometimes a bit overzealous with tests.

File: lib/smart_answer/calculators/property_fire_safety_payment_calculator.rb [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      @property_uprating_values ||= YAML.load_file(Rails.root.join("config/smart_answers/property_fire_safety_payment_data.yml")).freeze
+    end
+
+    def uprated_value_of_property
+      (property_uprating_values.fetch(@year_of_purchase, default_uprating_value) * @value_of_property.to_f).ceil
+    end
+
+    def fully_protected_from_costs?
+      uprated_value_of_property <= OUTSIDE_LONDON_VALUATION_LIMIT && live_in_london == "no" ||
+        uprated_value_of_property <= INSIDE_LONDON_VALUATION_LIMIT && live_in_london == "yes"
```
[the remark this answers is not in the document]
Comment (edwardkerry) [reply lib/smart_answer/calculators/property_fire_safety_payment_calculator.rb:34]: 👍

**Review by 1pretz1 — APPROVED** [review]
Sounds like a good idea! I'll hit the big green button 🍏

**Pull request merged**